core/src/main/scala/ru/itclover/tsp/core/aggregators/TimerPattern.scala

```diff
@@ -57,7 +57,7 @@ case class TimerAccumState[T](windowQueue: m.Queue[(Idx, Time)]) extends AccumSt
   private def createIdxValue(
     optStart: Option[(Idx, Time)],
     optEnd: Option[(Idx, Time)],
-    result: Result[Unit]
-  ): Option[IdxValue[Unit]] =
+    result: Result[Boolean]
+  ): Option[IdxValue[Boolean]] =
```

Comment:
why Boolean? We never return Result.succ(false), so there is the only possible returned value (true), which is effectively equivalent to Unit.

Reply:
The problem is that `Unit` does not automatically convert to `Boolean`, which is needed to perform complex operation like `Timer AND SomethingElse`.